File: WEB/labthree/src/main/java/utils/geometry/Point.java
package utils.geometry;

public class Point {
    final private Coordinate coordX;
    final private Coordinate coordY;

    public Point(double x, double y) {
        this.coordX = new DoubleCoordinate(x);
        this.coordY = new DoubleCoordinate(y);
    }
    
    public Point(double x, Coordinate coordY) {
        this.coordX = new DoubleCoordinate(x);
        this.coordY = coordY;
    }
    
    public Point(Coordinate coordX, double y) {
        this.coordX = coordX;
        this.coordY = new DoubleCoordinate(y);
    }
    
    public Point(Coordinate coordX, Coordinate coordY) {
        this.coordX = coordX;
        this.coordY = coordY;
    }
    
    public double x() {
        return coordX.get();
    }

    public double y() {
        return coordY.get();
    }
}

File: WEB/labfour/front/src/components/graph/Point.js
import React from "react";

export default function Point(props) {
    let className = `figure ${props.isHit ? 's' : 'f'}`
    return (
        <circle className={className} cx={600 + Math.round(props.coordX*100)} cy={600 - Math.round(props.coordY*100)} r="10"/>
    )
}